pkg/crane/manifest.go

```diff
@@ -14,11 +14,36 @@
 
 package crane
 
+import (
+	"fmt"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+)
+
 // Manifest returns the manifest for the remote image or index ref.
-func Manifest(ref string, opt ...Option) ([]byte, error) {
-	desc, err := getManifest(ref, opt...)
+func Manifest(r string, opt ...Option) ([]byte, error) {
+	o := makeOptions(opt...)
+	ref, err := name.ParseReference(r, o.name...)
+	if err != nil {
+		return nil, fmt.Errorf("parsing reference %q: %v", r, err)
+	}
+	desc, err := remote.Get(ref, o.remote...)
+	if err != nil {
+		return nil, err
+	}
+
+	if isIndex(desc) && o.platform == nil {
+		index, err := desc.ImageIndex()
+		if err != nil {
+			return nil, err
+		}
+		return index.RawManifest()
+	}
+
+	img, err := desc.Image()
 	if err != nil {
 		return nil, err
 	}
-	return desc.Manifest, nil
+	return img.RawManifest()
```

Comment:
This will still fail for schema 1 images, which we aren't able to satisfy the `v1.Image` interface for. If we keep the behavior the same here, but only try to resolve to an image when platform is specified, I think we can keep everything working, i.e.:

```go
if o.platform != nil {
  img, err := desc.Image()
  if err != nil {
    return nil, err
  }
  return img.RawManifest()
}

return desc.Manifest, nil
```

Sorry for all the back and forth on this, I really appreciate your contribution!